core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -140,32 +141,41 @@ public DockerClient client() {
                 "  Operating System: " + dockerInfo.getOperatingSystem() + "\n" +
                 "  Total Memory: " + dockerInfo.getMemTotal() / (1024 * 1024) + " MB");
 
-        String ryukContainerId = null;
         boolean useRyuk = !Boolean.parseBoolean(System.getenv("TESTCONTAINERS_RYUK_DISABLED"));
-        if (useRyuk) {
+        if (useRyuk && ryukContainerId == null) {
             ryukContainerId = ResourceReaper.start(hostIpAddress, client);
             log.info("Ryuk started - will monitor and terminate Testcontainers containers on JVM exit");
         }
 
         boolean checksEnabled = !TestcontainersConfiguration.getInstance().isDisableChecks();
         if (checksEnabled) {
-            VisibleAssertions.info("Checking the system...");
-            checkDockerVersion(version.getVersion());
-            if (ryukContainerId != null) {
-                checkDiskSpace(client, ryukContainerId);
-            } else {
-                runInsideDocker(
-                    client,
-                    createContainerCmd -> {
-                        createContainerCmd.withName("testcontainers-checks-" + SESSION_ID);
-                        createContainerCmd.getHostConfig().withAutoRemove(true);
-                        createContainerCmd.withCmd("tail", "-f", "/dev/null");
-                    },
-                    (__, containerId) -> {
-                        checkDiskSpace(client, containerId);
-                        return "";
-                    }
-                );
+            // fail-fast if checks have failed previously
+            if (cachedChecksFailure != null) {
+                throw cachedChecksFailure;
+            }
+
+            try {
+                log.info("Checking the system...");
+                checkDockerVersion(version.getVersion());
+                if (ryukContainerId != null) {
+                    checkDiskSpace(client, ryukContainerId);
+                } else {
+                    runInsideDocker(
+                        client,
+                        createContainerCmd -> {
+                            createContainerCmd.withName("testcontainers-checks-" + SESSION_ID);
+                            createContainerCmd.getHostConfig().withAutoRemove(true);
+                            createContainerCmd.withCmd("tail", "-f", "/dev/null");
+                        },
+                        (__, containerId) -> {
+                            checkDiskSpace(client, containerId);
+                            return "";
+                        }
+                    );
+                }
+            } catch (RuntimeException e) {
```

Comment:
WDYT about catching any `Exception`, not just `RuntimeException`?

Reply:
Well, unless there's a naughty `@SneakyThrows` somewhere, it doesn't look like any checked exceptions are thrown. Also, if we catch an `Exception` we're in a situation where the `client()` method would have to be able to thow `Exception`. We could _wrap_ a cached `Exception` inside a `RuntimeException`. But this feels a bit excessive.

Reply:
I am just cautious about the future changes where we introduce a checked exception that is thrown but not stored to be re-thrown later

Reply:
`@SneakyThrows` is another good point, yes

Reply:
Perhaps the good news is that a newly thrown checked exception would force us to catch it (client doesn't declare `throws`) - unless we use `@SneakyThrows` a lot more, which I think we should try and avoid. 😀